### DISCUSSION TITLE
iOS Launch Screen Issues

### DISCUSSION THREAD

**Discussion by MurrayHamilton** [airsdk/Adobe-Runtime-Support#1599] [Q&A]
I've only just got around to putting launch screens in my apps (better late than never :roll_eyes:) but I've come across a couple of issues. This could be because of me rather than a bug, so I thought I'd post in here rather than Issues.

My launch screen is extremely simple - it's just a View with a custom background colour, no images. The process of creating it was absolutely fine, and I've got LaunchScreen.storyboardc in the correct place with the correct reference. In a nutshell, it displays when I launch the app, but it's not behaving as I'd expect it to.

-----

_Some things to note:_
- _The app is compiled with AIR SDK 33.1.1.743_
- _The launch screen was created in Xcode 13.1_
- _The SWF has been built in release mode, not debug mode_
- _The app behaves the same in development and ad-hoc builds_
- _The videos below are screen captures from an actual iPhone 13 Pro Max running iOS 15.1.1._

-----

First of all, this is the app launching with the old-school PNG launch images - it gets to the main menu instantly and smoothly:

https://user-images.githubusercontent.com/14205929/151079331-c747417b-b78c-40c2-9f88-6c5dd2f1ee96.mp4

-----

This is the app launching with the launch screen only (no PNG launch images included while building the IPA) - notice the increase in time taken to get to the main menu (approximately 2 seconds) AND the flicker at about 4 seconds:

https://user-images.githubusercontent.com/14205929/151079330-21ede6d0-18d0-4a4b-a88a-9252e4fbf6fc.mp4

**_This video shows the flicker on an iPhone 7 Plus, it's a bit more pronounced:_**

https://user-images.githubusercontent.com/14205929/151084523-86fbccd7-c401-409f-9ee4-f9b2aef8e1c2.mp4

-----

This is the app launching with the launch screen AND the PNG launch images included - it still takes about 2 seconds to get to the main menu but the flicker has gone:

https://user-images.githubusercontent.com/14205929/151079327-fbacbf2b-99c0-4654-ac8c-021691d0b713.mp4

-----

My questions are:

1. Is it normal for the launch screen to increase launch time significantly, or have I inadvertently changed a setting somewhere in Xcode? It's pretty disappointing to go from an app that launches instantly to one that hangs for seemingly no reason.
2. Should I be including the old PNG launch images as well as the launch screen? My minimum iOS version is 9.0, so I thought I'd be OK to not include them.
3. If the answer to 2 is no, what can I try to stop the flicker? The SWF movie background, the main menu background and the launch screen are all the same colour.

Thanks in advance for any help :thumbsup:

**Comment by MurrayHamilton** [comment]
Sorry to be annoying, just giving this one bump as a last ditch attempt for any assistance!

**Reply by ajwfrost** [reply]
Sorry for the delay, was checking with the iOS folk here:
1. Yes, we actually add a delay deliberately so that people get a chance to see the launch screen...! That is mostly intended for the 'free tier' applications though, so I will see if there's a way we can eliminate the delay for the bespoke launch screens.
2. The old images do appear to make a difference as we've also seen that without them, the screen resolution does not always get set up properly. This may be behind the flicker as well (potentially the app launches but then has to wait for a 'resize' event). Sadly with the launch screen storyboard, we don't get any control over when the OS removes it so we may not be quite ready for the first frame rendering..
3. Not sure about this, it's a perennial problem on both iOS and Android. We had hoped to be able to pick up the stage colour (i.e. SWF background colour) and use that as a way to avoid obvious colour changes in the flicker, will see if we can get this working better...

thanks

**Reply by MurrayHamilton** [reply]
No need to apologise, I was being impatient! Thanks for getting back to me.

1. Ah right, that explains it! I understand why you've done that, but it would be really great to have no delay/an adjustable delay.
2. No worries, I'll just keep including the PNGs for now.

Thanks for your help!

**Reply by marchbold** [reply]
What should we be doing here exactly now @ajwfrost ? 

I had thought that including a launch storyboard negated the need for the default pngs unless you are targeting older iOS versions?

**Reply by ajwfrost** [reply]
Hi

Yes I think ideally we wouldn't need the launch images, we need to investigate a little more though about where the resolutions are coming from etc, this may be some legacy behaviour built into the AIR runtime which its using for convenience perhaps... we can investigate that part.

In terms of the flicker, @MurrayHamilton  can we please check:
- do you have the same images in your launch storyboard as are being used for the launch images? i.e. same set of resolutions?
- does this happen 100% of the time?
- if you can shut down all other running apps, then reboot the phone and start up so that it's in the best possible memory situation, does it still happen?

thanks

**Reply by MurrayHamilton** [reply]
- _**do you have the same images in your launch storyboard as are being used for the launch images? i.e. same set of resolutions?**_
I don't use any images in my launch screen that I'm aware of. All I did was change the background colour of the LaunchScreen view controller that Xcode generated with the project. This is what I have in Xcode: 
![xcode](https://user-images.githubusercontent.com/14205929/152387602-1cba2f6b-5204-42b4-8dbe-c117e6296e69.png)

The launch images I include are:

_iPhone ones_
> - Default~iphone (320x480)
> - Default@2x~iphone (640x960)
> - Default-568h@2x~iphone (640x1136)
> - Default-375w-667h@2x~iphone (750x1334)
> - Default-414w-736h@3x~iphone (1242x2208)
> - Default-812h@3x~iphone (1125x2436)
> - Default-896h@2x~iphone (828x1792)
> - Default-1242h@3x~iphone (1242x2688)

_iPad ones_
> - Default-Portrait~ipad (768x1024)
> - Default-Portrait@2x~ipad (1536x2048)
> - Default-Portrait-1112h@2x (1668x2224)
> - Default-Portrait-1194h@2x (1668x2388)
> - Default-Portrait@2x (2048x2732)

- _**does this happen 100% of the time?**_
Yes, if launch images are not included. As soon as I include the launch images the flicker is gone 100% of the time.

- _**if you can shut down all other running apps, then reboot the phone and start up so that it's in the best possible memory situation, does it still happen?**_
Yes

**Reply by MurrayHamilton** [reply]
@ajwfrost Sorry to dredge this back up, but I’m hoping to do an update for my apps soon. Is there anything I can do on my end to get rid of the launch screen delay?
